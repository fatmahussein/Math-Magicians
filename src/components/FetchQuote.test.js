import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import FetchQuote from './fetchQuote';

describe('quotes components', () => {
  test('renders loading state correctly', () => {
    const { asFragment } = render(<FetchQuote isLoading />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render loading while fetching data', () => {
    render(<FetchQuote />);
    expect(screen.getByText(/loading.../i)).toBeInTheDocument();
  });
});
