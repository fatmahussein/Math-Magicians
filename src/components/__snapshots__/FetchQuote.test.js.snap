// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`quotes components renders loading state correctly 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="author"
    >
      <p>
        Loading...
      </p>
    </div>
  </div>
</DocumentFragment>
`;
